modin/backends/pandas/query_compiler.py

```diff
@@ -1008,6 +1008,79 @@ def rolling_aggregate(self, rolling_args, func, *args, **kwargs):
         )
         return self.__constructor__(new_modin_frame)
 
+    def unstack(self, level, fill_value):
+        if not isinstance(self.index, pandas.MultiIndex) or (
+            isinstance(self.index, pandas.MultiIndex)
+            and is_list_like(level)
+            and len(level) == self.index.nlevels
+        ):
+            axis = 1
+            new_columns = ["__reduced__"]
+            need_reindex = True
+        else:
+            axis = 0
+            new_columns = None
+            need_reindex = False
+
+        def map_func(df):
+            return pandas.DataFrame(df.unstack(level=level, fill_value=fill_value))
+
+        is_all_multi_list = False
+        if (
+            isinstance(self.index, pandas.MultiIndex)
+            and isinstance(self.columns, pandas.MultiIndex)
+            and is_list_like(level)
+            and len(level) == self.index.nlevels
+            and need_reindex
+        ):
+            is_all_multi_list = True
+            real_cols_bkp = tuple(self.columns)
+            self.columns = [i for i in range(len(self.columns))]
```

Comment:
we don't want to change columns in place, also I believe that `np.arange(len(self.columns))` will work significantly faster in case of big amount of columns

Reply:
Is there another way to change `columns` for dataframe before computing via `apply_full_axis`? np.arange(len(self.columns)) was applied.

Reply:
`self.copy()`

Reply:
Is copying effective in case big dataframe?

Reply:
certainly not, but you don't want to change columns of that frame that passed into your function, you should either restore it at the end of the function (and also take about possible exception that could occur between changing `self.columns` and restoring it) or copy your frame

Reply:
`self.copy()` was added